sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamSuite.scala

```diff
@@ -1264,6 +1264,20 @@ class StreamSuite extends StreamTest {
       }
     }
   }
+
+  test("SPARK-34205: Pipe Streaming Dataset") {
+    assume(TestUtils.testCommandAvailable("cat"))
+
+    val inputData = MemoryStream[Int]
+    val piped = inputData.toDS()
+      .pipe("cat", (n, printFunc) => printFunc(n.toString)).toDF
+
+    testStream(piped)(
+      AddData(inputData, 1, 2, 3),
+      CheckAnswer(Row("1"), Row("2"), Row("3")),
+      AddData(inputData, 4),
+      CheckAnswer(Row("1"), Row("2"), Row("3"), Row("4")))
```

Comment:
I'd rather check `CheckNewAnswer(Row("4"))` to ensure inputs in previous batch are not affected to the next batch.

Reply:
Modified. Thanks.